### PR TITLE
Include skip steps in preparedutchf3.py

### LINKED ISSUE #9: Data ablation study capability: allow skipping of sections
Section slices could have quite a bit of duplicated data. 
To shorten training time you could split every other (or even more) sections.

### PR DESCRIPTION
**Pull request by squassina**
Updated code to allow skipping n lines to patch and section for training.

This will close issue #9 

I'm working in issue #10, but I'd like to have some revisions while I'm on this code.

### CODE REVIEW

File: scripts/prepare_dutchf3.py [hunk 1]
```diff
@@ -9,49 +9,59 @@
...
-def _get_aline_range(aline, per_val):
-    # Inline sections
-    test_aline = math.floor(aline * per_val / 2)
-    test_aline_range = itertools.chain(range(0, test_aline), range(aline - test_aline, aline))
-    train_aline_range = range(test_aline, aline - test_aline)
-
-    return train_aline_range, test_aline_range
+def _get_aline_range(aline, per_val, slice_steps):
+    try:
+        if slice_steps < 0:
```
Comment (olgaliak) [inline scripts/prepare_dutchf3.py]: is slice_steps==0 a valid case? It should be >=1, right?

File: scripts/prepare_dutchf3.py [hunk 2]
```diff
@@ -9,49 +9,59 @@
...
-        per_val (float, optional): the fraction of the volume to use for validation.
-            Defaults to 0.2.
+        output_dir (str): directory under data_dir to store the split files
+        label_file (str): npy files with labels. Stored in data_dir
+        stride (int): stride to use when sectioning of the volume
+        patch_size (int): size of patch to extract
+        per_val (float, optional):  the fraction of the volume to use for
+            validation. Defaults to 0.2.
+        log_config (str): path to log configurations
+        slice_steps (int): number of slices to be skipped. Defaults to 1.
```
Comment (olgaliak) [inline scripts/prepare_dutchf3.py]: Probably more detailed comment on how\why to use slice_steps could be used.
Also if read this comment as is, I  may get an impression that by default we are always skipping 1 (always skipping every other slice) but by default we should NOT skip anything.

File: scripts/prepare_dutchf3.py [hunk 3]
```diff
@@ -222,29 +277,33 @@ def split_alaudah_et_al_19(data_dir, stride, fraction_validation=0.2, loader_typ
         x_list = ["x_" + str(x) for x in range(xline)]
     elif loader_type == "patch":
         i_list = []
-        horz_locations = range(0, xline - stride, stride)
-        vert_locations = range(0, depth - stride, stride)
+        horz_locations = range(0, xline - patch_size + 1, stride)
+        vert_locations = range(0, depth - patch_size + 1, stride)
```
Comment (olgaliak) [inline scripts/prepare_dutchf3.py:274]: This method is marked with " DON'T USE, SEE NOTES BELOW" so I guess we should not be updating it. @maxkazmsft ,  I suggest to drop this method as it might be confusing.

File: scripts/prepare_dutchf3.py [hunk 4]
```diff
@@ -105,66 +132,90 @@ def split_patch_train_val(data_dir, stride, patch, per_val=0.2, log_config=None)
...
+                                                               slice_steps)
+    except RuntimeError as re:
+        logger.error(re)
 
     # Generate patches from sections
     # Process inlines
-    horz_locations = range(0, xline - stride, stride)
-    vert_locations = range(0, depth - stride, stride)
+    horz_locations = range(0, xline - patch_size + 1, stride)
+    vert_locations = range(0, depth - patch_size + 1, stride)
```
Comment (olgaliak) [inline scripts/prepare_dutchf3.py:153]: How does this change is related to  skip_steps?  Should it be part of a separate PR?

File: tests/test_prepare_dutchf3.py [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+        # calling the main function of the script without SLICE_STEPS, to check default value
+        prep_dutchf3.split_patch_train_val(data_dir=tmpdirname, output_dir=output, label_file=label_file, 
+        slice_steps=SLICE_STEPS, stride=STRIDE, patch_size=PATCH, per_val=PER_VAL,log_config=LOG_CONFIG)
+
+        # reading the file and splitting the data
+        patch_train = pd.read_csv(output + '/patch_train.txt', header=None, names=['row', 'a', 'b'])
+        patch_train = pd.DataFrame(patch_train.row.str.split('_').tolist(), columns=['aline', 'x', 'y', 'z'])
+
+        # test
+        # test
```
Comment (olgaliak) [inline tests/test_prepare_dutchf3.py]: dup comment?

**Review by olgaliak — CHANGES_REQUESTED** [review]
Thanks for putting the PR together, Lee!

File: scripts/prepare_dutchf3.py (hunk 4 above)
Comment (squassina) [reply scripts/prepare_dutchf3.py:153]: This was resolved in @kirasoderstrom PR and I incorporated here so I don't have conflicts once it's merged to the MSFT DeepSeismic repo.

File: scripts/prepare_dutchf3.py (hunk 3 above)
Comment (squassina) [reply scripts/prepare_dutchf3.py:274]: Are we going to open an issue & PR just to drop this method? If it should be dropped, I could open an issue and link that to this PR. Do you agree?

File: scripts/prepare_dutchf3.py (hunk 1 above)
Comment (squassina) [reply scripts/prepare_dutchf3.py]: Olga, isn't x < 0 the same as x >=1? I think the issue was with the ValueError message after this: "slice_steps cannot be a negative number". The message should be: "slice_steps cannot be zero or a negative number".

File: scripts/prepare_dutchf3.py (hunk 2 above)
Comment (squassina) [reply scripts/prepare_dutchf3.py]: Olga, I updated the documentation regarding that. 0 will not move the pointer and 1 means it will go to the next occurrence. 2 will skip every other slice and so on. The equation is SKIP = STEP - 1

The new message in doc should be: "slice_steps (int): increment to the slices count. If slice_steps > 1 the function will skip slice_steps - 1 slices. Defaults to 1, do not skip any slice."

File: tests/test_prepare_dutchf3.py (hunk 5 above)
Comment (squassina) [reply tests/test_prepare_dutchf3.py]: :) removed duplicated line.
